src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -0,0 +1,426 @@
+                    (registration, idx) -> registration.getAlternativeDataSharingPlanTargetDeliveryDate()),
+            new DatasetPropertyExtractor(
+                    "Alternative Data Sharing Plan Target Public Release Date", "alternativeDataSharingPlanTargetPublicReleaseDate", DatasetPropertyType.Date,
+                    (registration, idx) -> registration.getAlternativeDataSharingPlanTargetPublicReleaseDate()),
+            new DatasetPropertyExtractor(
+                    "Alternative Data Sharing Plan Controlled Open Access", "alternativeDataSharingPlanControlledOpenAccess", DatasetPropertyType.String,
+                    (registration, idx) -> {
+                        if (Objects.nonNull(registration.getAlternativeDataSharingPlanControlledOpenAccess())) {
+                            return registration.getAlternativeDataSharingPlanControlledOpenAccess().value();
+                        }
+                        return null;
+                    }),
+            new DatasetPropertyExtractor(
+                    "Data Location", "consentGroup.dataLocation", DatasetPropertyType.Json,
+                    (registration, idx) -> {
+                        if (Objects.nonNull(registration.getConsentGroups().get(idx).getDataLocation())) {
+                            return GsonUtil.getInstance().toJson(registration.getConsentGroups().get(idx).getDataLocation().stream().map(DataLocation::value).toList());
+                        }
+                        return null;
+                    }),
+            new DatasetPropertyExtractor(
+                    "File Types", "consentGroup.fileTypes", DatasetPropertyType.Json,
+                    (registration, idx) -> {
+                        if (Objects.nonNull(registration.getConsentGroups().get(idx).getFileTypes())) {
+                            return GsonUtil.getInstance().toJson(registration.getConsentGroups().get(idx).getFileTypes());
+                        }
+                        return null;
+                    }),
+            new DatasetPropertyExtractor(
+                    "URL", "consentGroup.url", DatasetPropertyType.String,
+                    (registration, idx) -> {
+                        if (Objects.nonNull(registration.getConsentGroups().get(idx).getUrl())) {
+                            return registration.getConsentGroups().get(idx).getUrl().toString();
+                        }
+                        return null;
+                    })
+    );
+
+
+    private List<DatasetProperty> convertRegistrationToDatasetProperties(DatasetRegistrationSchemaV1 registration, int consentGroupIdx) {
+        List<DatasetProperty> datasetProperties = new ArrayList<>();
+        for (DatasetPropertyExtractor datasetPropertyExtractor : datasetPropertyExtractors) {
+
+            Optional<DatasetProperty> extractedProperty = datasetPropertyExtractor.extract(registration, consentGroupIdx);
```

Comment:
you can do this in one line with lambdas:
```suggestion
            Optional<DatasetProperty> extractedProperty = datasetPropertyExtractor.extract(registration, consentGroupIdx)
            .ifPresent(datasetProperties::add);
```